spec/models/aggregated_cash_outgoings_spec.rb

```diff
@@ -0,0 +1,457 @@
+      let(:maintenance_out1) { create :cash_transaction, :credit_month_1, legal_aid_application: application, transaction_type: maintenance_out }
+      let(:maintenance_out2) { create :cash_transaction, :credit_month_2, legal_aid_application: application, transaction_type: maintenance_out }
+      let(:maintenance_out3) { create :cash_transaction, :credit_month_3, legal_aid_application: application, transaction_type: maintenance_out }
+
+      before do
+        maintenance_out1
+        maintenance_out2
+        maintenance_out3
+      end
+
+      it 'populates the model' do
+        aco = described_class.find_by(legal_aid_application_id: application.id)
+        expect(aco.check_box_maintenance_out).to eq 'true'
+        expect(aco.maintenance_out1).to eq maintenance_out1.amount
+        expect(aco.maintenance_out2).to eq maintenance_out2.amount
+        expect(aco.maintenance_out3).to eq maintenance_out3.amount
+        expect(aco.check_box_legal_aid).to be_nil
+        expect(aco.legal_aid1).to be_nil
+        expect(aco.legal_aid2).to be_nil
+        expect(aco.legal_aid3).to be_nil
+        expect(aco.check_box_rent_or_mortgage).to be_nil
+        expect(aco.rent_or_mortgage1).to be_nil
+        expect(aco.rent_or_mortgage2).to be_nil
+        expect(aco.rent_or_mortgage3).to be_nil
+        expect(aco.check_box_child_care).to be_nil
+        expect(aco.child_care1).to be_nil
+        expect(aco.child_care2).to be_nil
+        expect(aco.child_care3).to be_nil
+      end
+    end
+  end
+
+  describe '#valid?' do
+    before do
+      aco.update(params)
+    end
+
+    context 'valid params' do
+      context 'none selected' do
+        let(:params) { none_selected_params }
+
+        it 'is valid' do
+          expect(aco.valid?).to be true
+        end
+
+        it 'has the expected attributes' do
+          expect(aco.none_selected).to eq 'true'
+        end
+      end
+
+      context 'categories selected' do
+        let(:params) { valid_params }
+
+        it 'is valid' do
+          expect(aco.valid?).to be true
+        end
+
+        it 'has the expected attributes' do
+          expect(aco.check_box_rent_or_mortgage).to eq 'true'
+          expect(aco.rent_or_mortgage1).to eq '1'
+          expect(aco.rent_or_mortgage2).to eq '2'
+          expect(aco.rent_or_mortgage3).to eq '3'
+
+          expect(aco.check_box_maintenance_out).to eq 'true'
+          expect(aco.maintenance_out1).to eq '4'
+          expect(aco.maintenance_out2).to eq '5'
+          expect(aco.maintenance_out3).to eq '6'
+
+          expect(aco.none_selected).not_to be_present
+        end
+      end
+    end
+
+    context 'invalid params' do
+      context 'missing month value' do
+        let(:params) { missing_month_params }
+
+        it 'is invalid' do
+          expect(aco).not_to be_valid
+        end
+
+        it 'populates the errors' do
+          error_msg = 'Enter the total cash amount you received in each calendar month'
```

Comment:
received is income. paid is outgoings.

There are a few places on this test where wording needs changing: income should be outgoings and received should be paid etc.